tests/NimblePros.MediatR.Contrib.Test/Behaviors/CachingBehaviorTests.cs

```diff
@@ -0,0 +1,135 @@
+﻿using MediatR;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+
+namespace NimblePros.MediatR.Contrib.Test.Behaviors;
+
+public class CachingBehaviorTests
+{
+  [Fact]
+  public async Task Should_ReturnSuccess_GivenValidInputWithCacheMiss()
+  {
+    // Arrange
+    var nextMock = new Mock<INext>();
+    var memoryCacheMock = new Mock<IMemoryCache>();
+    var loggerMock = new Mock<ILogger<Mediator>>();
+    var cacheEntryMock = new Mock<ICacheEntry>();
+
+    var cachingBehavior = new CachingBehavior<TestCommand, bool>(memoryCacheMock.Object, loggerMock.Object);
+    var command = new TestCommand() { Name = "Test name" };
+
+    var cacheKey = "NimblePros.MediatR.Contrib.Test.Behaviors.Helpers.TestCommand";
+    object? result;
+
+    memoryCacheMock
+      .Setup(m => m.TryGetValue(cacheKey, out result))
+      .Returns(false);
+
+    memoryCacheMock
+      .Setup(m => m.CreateEntry(cacheKey))
+      .Returns(cacheEntryMock.Object);
+
+    nextMock
+      .Setup(m => m.Next())
+      .ReturnsAsync(await Task.FromResult(true));
```

Comment:
The mock setups are great and should be the main thing in each test's Arrange section after the other stuff is pulled up into fields.